play-json/shared/src/main/scala/JsMacroImpl.scala

```diff
@@ -49,7 +77,7 @@ import scala.reflect.macros.blackbox
    * @param matag The class of the reads/writes/format.
    * @param natag The class of the reads/writes/format.
    */
-  private def macroImpl[A, M[_], N[_], O <: Json.MacroOptions](methodName: String, mapLikeMethod: String, reads: Boolean, writes: Boolean)(implicit atag: c.WeakTypeTag[A], matag: c.WeakTypeTag[M[A]], natag: c.WeakTypeTag[N[A]], otag: c.WeakTypeTag[O]): c.Expr[M[A]] = {
+  private def macroImpl[A, M[_], N[_]](config: Tree, methodName: String, mapLikeMethod: String, reads: Boolean, writes: Boolean)(implicit atag: c.WeakTypeTag[A], matag: c.WeakTypeTag[M[A]], natag: c.WeakTypeTag[N[A]]): c.Expr[M[A]] = {
```

Comment:
Note: Replace type parameter `O` with parameter `config`

Reply:
Passing config as `c.Expr[JsonConfiguration]` ?

Reply:
`c.Expr[JsonConfiguration](inferImplicitValue ...)` advantage would "just" be to keep the type up to the Macro impl

Reply:
Ok I updated to use `c.Expr[JsonConfiguration]`